hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -477,6 +527,9 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
       ExplainCollector.setFilterStatement("none");
     }
 
+    SegmentStatusManager.ValidAndInvalidSegmentsInfo validAndInvalidSegmentsInfo =
+        DataMapUtil.getValidAndInvalidSegments(carbonTable, job.getConfiguration());
```

Comment:
Table status file should be read only once per query cycle, don't read multiple times

Reply:
passed the invalid segments